extensions/amp-truncate-text/1.0/amp-truncate-text.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CSS as COMPONENT_CSS} from './truncate-text.jss';
+import {CSS} from '../../../build/amp-truncate-text-1.0.css';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {TruncateText} from './truncate-text';
+import {dict} from '../../../src/utils/object';
+import {isExperimentOn} from '../../../src/experiments';
+import {userAssert} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-truncate-text';
+const EXPANDED_ATTR = 'i-amphtml-truncate-expanded';
+
+class AmpTruncateText extends PreactBaseElement {
+  /** @override */
+  init() {
+    this.registerApiAction('expand', (api) => api./*OK*/ expand());
+    this.registerApiAction('collapse', (api) => api./*OK*/ collapse());
+
+    return dict({
+      'onToggle': (isExpanded) => {
+        console.log(`toggle(${isExpanded})`);
+        if (isExpanded) {
+          this.element.setAttribute(EXPANDED_ATTR, '');
+        } else {
+          this.element.removeAttribute(EXPANDED_ATTR);
+        }
+      },
+    });
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    userAssert(
+      isExperimentOn(this.win, 'bento') ||
+        isExperimentOn(this.win, 'bento-truncate-text'),
+      'expected global "bento" or specific "bento-truncate-text" experiment to be enabled'
+    );
+    return super.isLayoutSupported(layout);
+  }
+}
+
+/** @override */
+AmpTruncateText['Component'] = TruncateText;
+
+/** @override */
+AmpTruncateText['children'] = {
```

Comment:
This definition is missing what to do with direct children. Also @dvoytenko this component is a goood example of one which needs the `passthrough` within the `children` def for text nodes.

Reply:
When I add the following block:
```json
{
  'children': {
    name: 'children',
    selector: '*',
    single: false,
  },
}
```
the `TextNode`(s) and `<p>` node get handled very differently (not picked up with the `*` selector) and don't slot as expected. Leaving the children definition out entirely was Justin's suggestion and seems to work correctly.

Reply:
Oh neat, TIL!

Reply:
I think we definitely want to cleanup this part of the bridge.

Reply:
/cc @jridgewell to loop in

Reply:
Did I say that? I thought we agreed on supporting a `children` key with a `name: ''` so it would act as the default slot? That way both Elements and Text nodes would be slotted into it.

Reply:
^^^ This is more likely, and I probably misunderstood. Will give that a try